### PR TITLE
Fixed "Debugging memory leaks with Guppy"( issue #4285)

### PR DESCRIPTION
**Pull request by sakshamb2113**
The section referring to guppy(for python 2.7) for debugging memory leaks now only refers to muppy(for python 3). Issue #4285.

### CODE REVIEW

**Comment by sakshamb2113** [comment]
@noviluni please review

**Comment by noviluni** [comment]
Hi @Gallaecio 

I've seen that after removing this we should probably delete these lines:

```
try:
    import guppy
    hpy = guppy.hpy()
except ImportError:
    hpy = None
```

from the `scrapy/extensions/telnet.py`. 

Do you agree?

File: scrapy/extensions/telnet.py [hunk 1]
```diff
@@ -110,7 +105,7 @@ def _get_telnet_vars(self):
             'est': lambda: print_engine_status(self.crawler.engine),
             'p': pprint.pprint,
             'prefs': print_live_refs,
-            'hpy': hpy,
+            'hpy': None,
```
Comment (Gallaecio) [inline scrapy/extensions/telnet.py]: I wonder what to do with this:
- Deprecate it somehow?
- Remove it?
- Use muppy?

In options 1 and 2, we might want to use muppy with a different name.

All of this is assuming muppy offers hpy or something similar, I’m not familiar with muppy or guppy at all.

Comment (sakshamb2113) [reply scrapy/extensions/telnet.py]: Can't we use muppy along with summary modules in pympler in place of guppy.hpy() as described in the documentation ?

**Comment by wRAR** [comment]
@sakshamb2113 can you please update this to resolve the conflict?

**Comment by sakshamb2113** [comment]
sure

**Comment by sakshamb2113** [comment]
@wRAR what should we do about the "hpy" key of telnet_vars in telnet.py ?

File: scrapy/extensions/telnet.py (hunk 1 above)
Comment (Gallaecio) [reply scrapy/extensions/telnet.py]: I think we should then just remove this key, instead of changing it to `None`.

**Pull request merged**